lua/telescope/builtin/git.lua

```diff
@@ -108,24 +109,42 @@ git.branches = function(opts)
 end
 
 git.status = function(opts)
-  local output = utils.get_os_command_output({ 'git', 'status', '-s' }, opts.cwd)
+  local gen_new_finder = function()
+    local output = utils.get_os_command_output({ 'git', 'status', '-s' }, opts.cwd)
 
-  if table.getn(output) == 0 then
-    print('No changes found')
-    return
+    if table.getn(output) == 0 then
+      print('No changes found')
+      return
+    end
+
+    return finders.new_table {
+      results = output,
+      entry_maker = opts.entry_maker or make_entry.gen_from_git_status(opts)
+    }
   end
 
+  local initial_finder = gen_new_finder()
+  if not initial_finder then return end
+
   pickers.new(opts, {
     prompt_title = 'Git Status',
-    finder = finders.new_table {
-      results = output,
-      entry_maker = make_entry.gen_from_git_status(opts)
-    },
+    finder = initial_finder,
     previewer = previewers.git_file_diff.new(opts),
     sorter = conf.file_sorter(opts),
     attach_mappings = function(_, map)
-      map('i', '<tab>', actions.git_staging_toggle)
-      map('n', '<tab>', actions.git_staging_toggle)
```

Comment:
Why did you want to remove these from `actions`? I'm OK with doing it, but maybe there is some way we can still keep these around.

Reply:
Its now a local function so we are able to generate a `new_finder`. Otherwise we don't have information about that finder. I can still keep it action around tho

Reply:
Maybe we can use `enhanced post` here. Just don't close telescope